app/src/main/java/de/westnordost/streetcomplete/StreetCompleteApplication.kt

```diff
@@ -21,18 +46,23 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.cancel
 import kotlinx.coroutines.launch
+import org.koin.android.ext.android.inject
+import org.koin.android.ext.koin.androidContext
+import org.koin.android.ext.koin.androidLogger
+import org.koin.androidx.workmanager.koin.workManagerFactory
+import org.koin.core.context.startKoin
+import org.koin.core.logger.Level
 import java.lang.System.currentTimeMillis
 import java.util.concurrent.TimeUnit
-import javax.inject.Inject
 
 class StreetCompleteApplication : Application() {
 
-    @Inject lateinit var preloader: Preloader
-    @Inject lateinit var crashReportExceptionHandler: CrashReportExceptionHandler
-    @Inject lateinit var resurveyIntervalsUpdater: ResurveyIntervalsUpdater
-    @Inject lateinit var downloadedTilesDao: DownloadedTilesDao
-    @Inject lateinit var prefs: SharedPreferences
-    @Inject lateinit var editHistoryController: EditHistoryController
+    private val preloader: Preloader by inject()
+    private val crashReportExceptionHandler: CrashReportExceptionHandler by inject()
+    private val resurveyIntervalsUpdater: ResurveyIntervalsUpdater by inject()
+    private val downloadedTilesDao: DownloadedTilesDao by inject()
+    private val prefs: SharedPreferences by inject()
+    private val editHistoryController: EditHistoryController by inject()
```

Comment:
And this works? Because `startKoin` is only called in `onCreate` but these fields are created when the application is created.

Reply:
I was also concerned about this, but looking into the `inject` method, it's [implemented as a lazy delegate](https://github.com/InsertKoinIO/koin/blob/main/android/koin-android/src/main/java/org/koin/android/ext/android/ComponentCallbackExt.kt#L40-L44) for Koins `get`.

I wonder if there's a way to make this less confusing. 🤔

Reply:
Ah okay. Well, a comment would resolve it, but nevermind. That class is touched not very often and if it works, it works.